### PR TITLE
feat(osc): add project list  adapter oms new api

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-onlineschemachange

#### What this PR does / why we need it:
oms api changed, so wo should adapter the new api

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package com.oceanbase.odc.service.onlineschemachange.oms.enums;
+
+/**
+ * @author yaobin
+ * @date 2023-12-27
+ * @since 4.2.4
+ */
+public enum ProjectTypeEnum {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java:24]: What is this "Project"? I mean it should be different from ODC's project, so better to add a prefix (Such as "Oms") to specify.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java [hunk 2]
```diff
@@ -30,6 +32,15 @@
  * @since 4.2.0
  */
 public interface ProjectOpenApiService {
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java:34]: Ditto.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java (hunk 1 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java:24]: Enum suffix is no need.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java [hunk 3]
```diff
@@ -30,6 +32,15 @@
...
+
+    /**
+     * 查询项目
+     *
+     * @param request
+     * @return
+     */
+    List<ProjectResponse> listProjects(ListProjectRequest request);
```
Comment (smallsheeeep) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java:42]: Is this interface currently unavailable anywhere?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/enums/ProjectTypeEnum.java:24]: okey, done, rename to OmsProjectType

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java:34]: this class suggest reserve, it's referenced by many class and name contain `OpenApi` which can be different from odc project

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java]: done , all osc class with project* rename to OmsProject*

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java (hunk 3 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oms/openapi/ProjectOpenApiService.java]: it's method be called in cloud project

**Review by smallsheeeep — APPROVED** [review]
LGTM

**Pull request merged**